frameworks/core_foundation/views/view/statechart.js

```diff
@@ -1511,7 +1511,10 @@ SC.CoreView.reopen(
   */
   _isVisibleDidChange: function () {
     if (this.get('isVisible')) {
-      this._doShow();
+      // show the view if it's not being shown already
+      if (this.get('viewState') !== SC.CoreView.ATTACHED_SHOWN) {
```

Comment:
There are actually a few different "shown" states. Can you make this into,

    if (!(this.get('viewState') & SC.CoreView.IS_SHOWN)) {

?